src/java_bytecode/java_entry_point.cpp

```diff
@@ -94,11 +94,6 @@ static void java_static_lifetime_init(
         if(allow_null)
         {
           irep_idt nameid=sym.symbol_expr().get_identifier();
-          std::string namestr=id2string(nameid);
```

Comment:
When can `Foo.class` field be `null`? If this is so, I don't think the above test demonstrates it as wasn't the assert already unreachable by the fact we assume `Class.field` is null (and we used to require it be non-null).

Reply:
It can't. At some point this was initialising a `java.lang.Class *`, when the `allow_null = false;` made sense. Now it's initialising a `java.lang.Class` (i.e. not a pointer), so it's no longer relevant (the `allow_null` only has an effect on `Class`'s fields, not the pointer itself, which is created as `&java.lang.Integer@class_model` (for example) as each reference site)